java/poi/src/main/java/com/example/App.java

```diff
@@ -0,0 +1,82 @@
+import java.sql.Connection;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Types;
+
+import com.ibm.as400.access.AS400JDBCDriver;
+import com.ibm.as400.access.AS400;
+
+import java.util.Properties;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+
+import java.lang.Exception;
+
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Cell;
+
+
+public class App
+{
+    public static void main( String[] args ) throws Exception
+    {
+      Properties props = new Properties();
+      props.load(new FileInputStream(".env"));
+
+      String HOST = props.getProperty("DB_HOST");
+      String USER = props.getProperty("DB_USER");
+      String PASS = props.getProperty("DB_PASS");
+      String query = "SELECT CUSNUM, LSTNAM, BALDUE, CDTLMT FROM QIWS.QCUSTCDT";
+
+      AS400JDBCDriver driver = new AS400JDBCDriver();
+      AS400 as400 = new AS400(HOST, USER, PASS);
+
+      Connection connection = driver.connect(as400);
+      Statement statement = connection.createStatement();
+      ResultSet results = statement.executeQuery(query);
+      ResultSetMetaData metadata = results.getMetaData();
+
+      // create workbook and workshee
+      Workbook workbook = WorkbookFactory.create(true);
+      Sheet sheet = workbook.createSheet("Customers");
+
+      // add the header
+      Row header = sheet.createRow(0);
+      int columnCount = metadata.getColumnCount();
+
+      for (int i = 0; i < columnCount; i++) {
+        // column name start with index 1
+        header.createCell(i).setCellValue(metadata.getColumnName(i + 1));
+      }
+
+      // fill in the worksheet
+      for (int r = 1; results.next(); r++) {
+        Row row = sheet.createRow(r);
+        for (int c = 0; c < columnCount; c++) {
+          Cell cell = row.createCell(c);
+          // column name and type start with index 1
+          String columnName = metadata.getColumnName(c + 1);
+          int columnType = metadata.getColumnType(c + 1);
+
+          switch (columnType) {
+            case Types.NUMERIC:
+              row.createCell(c).setCellValue(results.getDouble(columnName));
+            break;
+            case Types.CHAR:
+              row.createCell(c).setCellValue(results.getString(columnName));
+            break;
+          }
+        }
+      }
+
+      // write the xlsx file
+      workbook.write(new FileOutputStream("Customers.xlsx"));
+      System.out.println("Succesfully created Customers.xlsx!");
+      connection.close();
+      System.exit(0);
```

Comment:
An explicit `System.exit(0)` shouldn't be needed. If it hangs without this, there's some other problem we should look into

Reply:
So I noticed on my linux machine when a user is not specified a GUI window pops up prompting for credentials:


![image](https://user-images.githubusercontent.com/33973272/92155674-5c4fff00-eded-11ea-9765-0cebddd36f8a.png)


Without adding the the System.exit line it hangs for some reason. If you specify the creditials using .env file there is no hang. Also when ran on IBM i there is no hang because the current user is used.

Reply:
@abmusse we should submit this as a bug to JTOpen. I'm guessing the problem is that they aren't using `DISPOSE_ON_CLOSE` ([doc](https://docs.oracle.com/javase/7/docs/api/javax/swing/WindowConstants.html#DISPOSE_ON_CLOSE)) as the default close operation on that window.

Reply:
Regardless, you can turn that dialog off with [this function](https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_73/rzahh/javadoc/com/ibm/as400/access/AS400.html?view=kc#setGuiAvailable(boolean))

Reply:
Nice I will disable the gui so that the behavior is the same when run non any platform.